ServerCore/Pages/Threads/PuzzleThread.cshtml

```diff
@@ -0,0 +1,149 @@
+﻿@page "/{eventId}/{eventRole}/Threads/PuzzleThread/{puzzleId}"
+@using ServerCore.DataModel
+@model ServerCore.Pages.Threads.PuzzleThreadModel
+
+@{
+    ViewData["Title"] = "Puzzle thread";
+    ViewData["AdminRoute"] = "/Threads/PuzzleThread";
+    ViewData["AuthorRoute"] = "/Threads/PuzzleThread";
+    ViewData["PlayRoute"] = "/Threads/PuzzleThread";
+    ViewData["RoutingPuzzleId"] = ViewContext.RouteData.Values["puzzleId"];
+}
+<script>
+    function EditText(id) {
+        let textAreaId = "TextAreaEdit_" + id;
+        let textAreaEditBox = document.getElementById(textAreaId);
+        textAreaEditBox.classList.remove("hidden");
+        let originalTextId = "MessageText_" + id;
+        let originalText = document.getElementById(originalTextId);
+        originalText.classList.add("hidden");
+    }
+
+    function CancelEdit(id) {
+        let textAreaId = "TextAreaEdit_" + id;
+        let textAreaEditBox = document.getElementById(textAreaId);
+        textAreaEditBox.classList.add("hidden");
+        let originalTextId = "MessageText_" + id;
+        let originalText = document.getElementById(originalTextId);
+        originalText.classList.remove("hidden");
+    }
+</script>
+<h2>Help thread</h2>
+<div>
+    @if (Model.EventRole == ServerCore.ModelBases.EventRole.play)
+    {
+        <a asp-page="/Submissions/Index" asp-route-puzzleId="@Model.Puzzle.ID">Back to puzzle</a>
+    }
+    else
+    {
+        <a asp-page="/Threads/PuzzleThreads">Back to thread list</a>
+    }
+</div>
+
+<h4>Thread for puzzle @Model.Puzzle.Name</h4>
+<hr />
+@foreach (var message in Model.Messages)
+{
+    <div class="message-container">
```

Comment:
You deleted the message-game-control class here, but the CSS rule still exists. Accidental deletion or unused rule?